Kudu.Core/Deployment/Generator/FunctionAppBuilder.cs

```diff
@@ -2,18 +2,18 @@
 
 namespace Kudu.Core.Deployment.Generator
 {
-    class FunctionAppBuilder : BaseBasicBuilder
+    class FunctionAppBuilder : MicrosoftSiteBuilder
     {
-        public FunctionAppBuilder(IEnvironment environment, IDeploymentSettingsManager settings, IBuildPropertyProvider propertyProvider, string repositoryPath, string projectPath) 
-            :base(environment, settings, propertyProvider, repositoryPath, projectPath, "--functionApp")
+        public FunctionAppBuilder(IEnvironment environment, IDeploymentSettingsManager settings, IBuildPropertyProvider propertyProvider, string sourcePath, string projectFilePath, string solutionPath)
+            : base(environment, settings, propertyProvider, sourcePath, projectFilePath, solutionPath, "--functionApp")
         {
         }
 
         public override string ProjectType
         {
             get
             {
-                return "functionApp";
+                return "functionApp Msbuild";
```

Comment:
Let's call this 'COMPILED FUNCTIONAPP'. (seems all the existing one are upper case)